layers.go

```diff
@@ -527,7 +527,11 @@ func (r *layerStore) SetMetadata(id, metadata string) error {
 }
 
 func (r *layerStore) tspath(id string) string {
-	return filepath.Join(r.layerdir, id+tarSplitSuffix)
+	tmpPath := ""
+	if DefaultStoreOptions.GraphRoot == "" {
+		tmpPath = filepath.Join(DefaultStoreOptions.GraphRoot, "tmp")
+	}
+	return filepath.Join(tmpPath, r.layerdir, id+tarSplitSuffix)
 }
```

Comment:
Why is this moving the tarsplit data to a directory named "tmp"?  We need that information when reading layers, so we don't want to suggest that it's okay to discard it.